### ISSUE TITLE
[BUG] Incorrect telemetry timestamp when mobile device wakes up

### ISSUE THREAD

**Issue by gabrielweyer** [microsoft/ApplicationInsights-JS#1571]
**Description/Screenshot**

We have a [Next.js](https://nextjs.org/) application mainly used on mobile devices. This frontend application is issuing HTTP requests to APIs that are not hosted in Azure. The APIs are not instrumented with Application Insights but the frontend application is. While investigating an issue I discovered that for some requests Application Insights was reporting a timestamp that was 20 minutes ahead of the timestamp reported by the backend. I was able to reproduce the issue on my iPhone.

1. I loaded the app on my iPhone in Safari and triggered an API request
2. I alt-tabbed, loaded another website and locked my device
3. Roughly 30 minutes later I unlocked my device, alt-tabbed to the app in Safari (without reloading the page) and triggered another API request

| Request issued at | Time recorded by App Insights | Time recorded by backend telemetry |
| - | - | - |
| 1:18:43 PM | 1:18:43 PM | 1:18:43 PM |
| - | Device is locked | - |
| 1:51:55 PM | 1:31:01 PM | 1:51:55 PM |

After waking up the device, the first API request was recorded with a timestamp 21 minutes in the past.

We're using `xhr` for the API requests.

**Steps to Reproduce**

 - OS/Browser: iPhone SE, iOS 14.4.2, Mobile Safari 14.0.3
 - SDK Version [e.g. 22]:
    -  `@microsoft/applicationinsights-react-js`: `3.0.5`
    - `@microsoft/applicationinsights-web`: `2.6.0`
 - How you initialized the SDK:

```javascript
import {ApplicationInsights} from '@microsoft/applicationinsights-web';
import {ReactPlugin} from '@microsoft/applicationinsights-react-js';

let reactPlugin = null;
let appInsights = null;

/**
 * Create the App Insights Telemetry Service
 *
 * Taken from: https://github.com/Azure-Samples/application-insights-react-demo/blob/f8c7f39e156c8b8d116332b9714ce400d2261f4f/src/TelemetryService.js
 *
 * @return {{reactPlugin: ReactPlugin, appInsights: Object, initialize: Function}} - Object
 */
const createTelemetryService = () => {

    /**
     * Initialize the Application Insights class
     * @param {string} applicationVersion - Application version, usually a release or build number
     * @param {string} instrumentationKey - Application Insights Instrumentation Key
     * @param {Object} browserHistory - client's browser history, supplied by the withRouter HOC
     * @return {void}
     */
    const initialize = (applicationVersion, instrumentationKey, browserHistory) => {
        if (!applicationVersion) {
            throw new Error('Application version not provided when initialising TelemetryService.');
        }

        if (!instrumentationKey) {
            throw new Error('Application Insights instrumentation key not provided when initialising TelemetryService.');
        }

        if (!browserHistory) {
            throw new Error('history not provided when initialising TelemetryService.');
        }

        reactPlugin = new ReactPlugin();

        appInsights = new ApplicationInsights({
            config: {
                instrumentationKey: instrumentationKey,
                disableFetchTracking: false,
                extensions: [reactPlugin],
                extensionConfig: {
                    [reactPlugin.identifier]: {
                        history: browserHistory
                    }
                }
            }
        });

        appInsights.loadAppInsights();
        var telemetryInitializer = (envelope) => {
            /*
             * Keys were sourced from:
             * - https://github.com/microsoft/ApplicationInsights-JS/blob/6a9259c0c548c947825b9f0b084839b24fac5f7d/README.md#example-setting-cloud-role-name
             * - https://github.com/microsoft/ApplicationInsights-JS/blob/88cf8edccfefd7d4bcd1657f15715076b83bf325/shared/AppInsightsCommon/src/Interfaces/Contracts/Generated/ContextTagKeys.ts#L212-L272
             */

            envelope.tags['ai.application.ver'] = applicationVersion;
            envelope.tags['ai.cloud.role'] = 'redacted-name';
          };
        appInsights.addTelemetryInitializer(telemetryInitializer);
    };

    return {reactPlugin, appInsights, initialize};
};

export const ai = createTelemetryService();
export const getAppInsights = () => appInsights;

```

**Expected behavior**

We're expecting the telemetry items to be recorded with the correct timestamp.

**Additional context**

**Comment by MSNev** [comment]
Timestamps are stamped on events at the point of the event being recorded (track getting called) effectively using the DateTime.now(), when devices hibernate apps different environments handle the clock skew that occurs differently and as a JS runtime we cannot control the actual timestamp reported by the environment (ie. we can't and don't want to perform our own NTP lookup).

The server has a similar issue it that it can't control the time from all devices and clock skew is a common issue, as such it records the only time it can guarantee -- the time the event is received.

The issue you describe above is either
- The request sent @ 1:51 includes events that where "batched" (track was called but no request was sent), while the device was locked around 1:31 (assumes the app was not 100% hibernated)
- The event @ 1:31 was triggered on "wake up" and the safari / iOS runtime was "speeding up time" to catchup, this is one way that some JS runtimes re-sync the time (not sure if that is the case this time -- just a general comment from my previous work on/with mobile devices / apps) vs simply "jumping" to the current time (this later can cause some apps issue -- especially with timer execution and cron style background processing)

In short, I don't believe that there is anything the SDK (as a JS runtime) can do to fix or avoid this scenario.
